smart_card_connector_app/src/chrome-api-provider-unittest.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+goog.require('GoogleSmartCard.ConnectorApp.ChromeApiProvider');
+goog.require('goog.Promise');
+goog.require('goog.testing');
+goog.require('goog.testing.MockControl');
+goog.require('goog.testing.PropertyReplacer');
+goog.require('goog.testing.StrictMock');
+goog.require('goog.testing.jsunit');
+goog.require('goog.testing.mockmatchers');
+goog.require('goog.testing.messaging.MockMessageChannel');
+
+goog.setTestOnly();
+
+goog.scope(function() {
+
+const GSC = GoogleSmartCard;
+const ChromeApiProvider = GSC.ConnectorApp.ChromeApiProvider;
+
+const propertyReplacer = new goog.testing.PropertyReplacer();
+/** @type {!goog.testing.MockControl|undefined} */
+let mockControl;
+let mockServerMessageChannel;
+
+goog.exportSymbol('testChromeApiProvider', {
+  'setUp': function() {
+    mockControl = new goog.testing.MockControl();
+    mockServerMessageChannel =
+        new goog.testing.messaging.MockMessageChannel(mockControl);
+    for (const event
+             of ['onEstablishContextRequested',
```

Comment:
nit: Let's create a constant with this list at the top of the file. `EXTENSION_API_EVENTS` or smth like that.